.github/workflows/check-release.yml

```diff
@@ -0,0 +1,202 @@
+name: Check Release and Update Dockerfile
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * *' # Execute everyday at 9 AM
+
+env:
+  CURRENT_TAG: ${{ vars.CURRENT_TAG }}
+  CURRENT_SHA: ${{ vars.CURRENT_SHA }}
+
+jobs:
+  check-release:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        name: Checkout
+    
+      - name: Get latest release from another repo
+        id: get_release
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const latestRelease = await github.rest.repos.getLatestRelease({
+              owner: 'actions',
+              repo: 'runner'
+            });
+            const releaseInfo = {
+              tag_name: latestRelease.data.tag_name,
+              sha: latestRelease.data.target_commitish,
+            };
+            console.log(JSON.stringify(releaseInfo));
+            return JSON.stringify(releaseInfo);
+      
+      - name: Define release outputs
+        id: release_outputs
+        run: |
+          release_info="${{ steps.get_release.outputs.result }}"  
+          tag_name=$(echo "$release_info" | jq -r '.tag_name')
+          sha=$(echo "$release_info" | jq -r '.sha')
+          short_tag_name=$(echo "$tag_name" | sed 's/^v//')
+
+          echo "tag_name=$tag_name" >> "$GITHUB_OUTPUT"
+          echo "sha=$sha" >> "$GITHUB_OUTPUT"
+          echo "short_tag_name=$short_tag_name" >> "$GITHUB_OUTPUT"
+      
+      # Check if the release has changed
+
+      - name: Check if release has changed
+        id: check_release_changes
+        run: |
+          new_tag=${{ steps.release_outputs.outputs.tag_name }}
+          new_sha=${{ steps.release_outputs.outputs.sha }}
+
+          current_tag=${{ env.CURRENT_TAG }}
+          current_sha=${{ env.CURRENT_SHA }}
+
+          if [[ "$current_tag" == "$new_tag" && "$current_sha" == "$new_sha" ]]; then
+            echo "No changes in release."
+            echo "release_changed=false" >> "$GITHUB_OUTPUT"
+          else
+            echo "Release has changed."
+            echo "release_changed=true" >> "$GITHUB_OUTPUT"
+
+            # Update repository variables
+            gh variable set CURRENT_TAG \
```

Comment:
if we can't use ${{ secrets.GITHUB_TOKEN }} I suggest to write this information in a file so we reduce permission scopes

Reply:
Personally, I prefer to not have repository changes through pipelines.
For instance, what does it happen if a PR is not merged in the same day it is opened? I think we would end up having a pull request for each day until one of them is merged, as the workflow always compare the versioned file with what is found on the runner repository.

Using repository/environment variable allows us to update that value when the first PR is opened, avoiding clones.

Unfortunately, we need a PAT as `secrets.GITHUB_TOKEN` is not enough (GitHub permission model is a shame), and I see the bot as the best way to implement it (if you can think of a better solution I'm open). *Unfortunately*, this is not the first time we have to use bots..

Reply:
I agree with @Krusty93, with the file as long as it is not merged the workflow will continue to find a release to update. 
But if we want to use a file the only alternative that comes to mind is to do a step that checks the existence of the branch `update-release-to-${{ steps.release_outputs.outputs.tag_name }}` which is normally created if there is an update to be done. 
In this way you can have a control similar to that of environment variables.

Reply:
In my review this is the last open point.
I remember that the write access can modify the secrets. Can we try to use secrets instead of variables?

Reply:
It can’t be done due to insufficient permissions granted to the job (also `write-all` is insufficient), but this might be unnecessary. It doesn’t matter if that value is visible to everyone. It’s simply a check to identify the resource currently in use, and if it gets altered, it would only result in a PR that doesn’t need to be merged.

Reply:
So I think that suggested to checks the existence branch with naming convention can be a good tradeoff

`update-release-to-${{ steps.release_outputs.outputs.tag_name }}`